virttest/qemu_storage.py

```diff
@@ -1550,10 +1550,49 @@ def measure(self, target_fmt, size=None, output="human"):
         cmd_list = [self.image_cmd, "measure", ("--output=%s" % output),
                     ("-O %s" % target_fmt)]
         if size:
+            if self._secret_objects:
+                cmd_list.extend(self._secret_objects)
+                secret_object_id = self.encryption_config.key_secret.aid
+                cmd_list.append("-o key-secret=%s" % secret_object_id)
             cmd_list.append(("--size %s" % size))
         else:
-            cmd_list.extend([("-f %s" % self.image_format),
-                             self.image_filename])
+            if self.encryption_config.key_secret:
```

Comment:
The raw process:

1. target image
The following part can be moved out of else, for when size is given, we still have to define the secret objects for target, also this part might be placed before 'if size'
   # the target image format is luks
   if target_fmt == 'luks':
        target_image_object = QemuImg()
        cmd_Llist.append(target_image_object._secret_objects[-1])
        cmd_list.append('-o key-secret=%s' %  target_image_object.encryption_config.key_secret.aid)


2. the source image, inside else code block
  # the image itself is in luks format
  if self.encryption_config.key_secret:
      cmd_list.append(self._secret_objects[-1])
      cmd_list.append("'%s'" % get_image_json(self.tag, self.params, self.root_dir))
  else:
      cmd_list.extend([("-f %s" % self.image_format), self.image_filename])


Then we don't need 'if elif elif else' code blocks

Please be noted that this is the basic process I think, there may be some details that I didn't consider yet. Just FYI.

Reply:
Agree with you, that makes the code cleaner. I will update them. Many thanks.